query-engine/connectors/mongodb-query-connector/src/orderby.rs

```diff
@@ -210,21 +261,22 @@ impl OrderByBuilder {
             // Unwind order by aggregate to-one middle joins into the top level join
             // to prevent nested join result which break the stages that come after
             // See `unwind_aggregate_joins` for more explanation
-            if let OrderBy::Aggregation(order_by_aggregate) = &data.order_by {
+            if let OrderBy::ToManyAggregation(order_by_aggregate) = &data.order_by {
                 if !order_by_aggregate.path.is_empty() {
                     match order_by_aggregate.sort_aggregation {
                         prisma_models::SortAggregation::Count => {
-                            if data.order_by.has_middle_to_one_path() {
+                            if let Some(clone_to) = data.prefix.as_ref().and_then(|x| x.clone_to.clone()) {
+                                order_aggregate_proj_doc.push(doc! { "$addFields": { clone_to.clone(): { "$size": { "$ifNull": [format!("${}", data.full_reference_path(false)), []] } } } });
+                                field_name = clone_to; // Todo: Just a hack right now, this whole function needs love.
```

Comment:
Do we have a ticket for clean up work we want to do once we have launched mongo and have a little more time for clean up?

Reply:
Let's just track tech debt tickets in the engine repo directly with "tech debt" label attached: https://github.com/prisma/prisma-engines/issues/2716